lib/Doctrine/ODM/MongoDB/UnitOfWork.php

```diff
@@ -919,7 +919,8 @@ private function computeAssociationChanges($parentDocument, $mapping, $value)
                     $this->scheduleCollectionUpdate($value);
                 }
             }
-            $this->visitedCollections[] = $value;
+            // Inline call to UnitOfWork::markCollectionVisited()
+            $this->visitedCollections[spl_object_hash($value)] = $value;
```

Comment:
@jmikola @malarzm that's easy. We just cannot use `SplObjectStorage` here because you require the object when fetching.
When you look at its API, you have this:

```php
class SplObjectStorage
{
    function offsetExists ( object $object );
    function offsetGet ( object $object );
    function offsetSet ( object $object [, mixed $data = NULL ] );
    function offsetUnset ( object $object );

    // ...
}
```

This means if you use the class, you must have `$value` to retrieve... ?